src/ex_cmds.c

```diff
@@ -1531,26 +1531,47 @@ make_filter_cmd(
 {
     char_u	*buf;
     long_u	len;
-
-#if defined(UNIX)
-    int		is_fish_shell;
     char_u	*shell_name = get_isolated_shell_name();
+    int		is_powershell;
+#ifdef UNIX
+    int		is_fish_shell;
+#endif
 
     if (shell_name == NULL)
 	return NULL;
 
+#if defined(UNIX)
     // Account for fish's different syntax for subshells
     is_fish_shell = (fnamecmp(shell_name, "fish") == 0);
-    vim_free(shell_name);
     if (is_fish_shell)
 	len = (long_u)STRLEN(cmd) + 13;		// "begin; " + "; end" + NUL
     else
 #endif
-	len = (long_u)STRLEN(cmd) + 3;			// "()" + NUL
+    {
+	is_powershell = (shell_name[0] == 'p')
+			&& (fnamecmp(shell_name, "powershell") == 0
+				|| fnamecmp(shell_name, "powershell.exe") == 0
+				|| fnamecmp(shell_name, "pwsh") == 0
+				|| fnamecmp(shell_name, "pwsh.exe") == 0);
+	if (is_powershell)
+	    len = (long_u)STRLEN(cmd) + 3;
+	else
+	    len = (long_u)STRLEN(cmd) + 3;			// "()" + NUL
+    }
+
     if (itmp != NULL)
-	len += (long_u)STRLEN(itmp) + 9;		// " { < " + " } "
+    {
+	if (is_powershell)
+	    // "& { Get-Content" + " | & " + " }"
+	    len += (long_u)STRLEN(itmp) + 23;
```

Comment:
There is a space after `Get-Content` (see L1603). len should be 24, I think.

Reply:
Changed the value to 24.